codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/EventStreamUnmarshallerGenerator.kt

```diff
@@ -339,6 +340,7 @@ class EventStreamUnmarshallerGenerator(
                     // TODO(EventStream): Errors on the operation can be disjoint with errors in the union,
                     //  so we need to generate a new top-level Error type for each event stream union.
                     when (codegenTarget) {
+                        // TODO(https://github.com/awslabs/smithy-rs/issues/1970) It should be possible to unify these branches now
```

Comment:
Why should it be possible?

Reply:
because we have the builderInstantiator, I think. But it's possible this is more divergence that can't be handled like that.